include/deal.II/base/task_result.h

```diff
@@ -238,10 +238,72 @@ namespace Threads
      * result are not put into the current object, it still means that that task
      * is working on data that is changing as it is working, with obviously
      * unpredictable results.)
+     *
+     * @note As mentioned above, it is a considered a bug to assign a task to
+     *   a TaskResult object that already has a task running. That means that
+     *   you will get in trouble if multiple threads (or multiple tasks running
+     *   concurrently) call this operator at the same time: One of these
+     *   threads will set a task, and the other threads will try the same but
+     *   because the background task is likely still running will encounter
+     *   an error. As a consequence, you cannot easily use this operator
+     *   from multiple threads. Use try_emplace_task() in that case.
```

Comment:
Can you detect this situation and report an error?

Reply:
Yes, in fact that is what `operator=` does. Sort of at least. All we can check is whether the other task is still running, and then we report the error. We cannot check whether the other task has finished and the user has waited for it (as they should), or whether the other task just happened to have already finished by the time we got there. So we can't deterministically throw the error (you'd need a race condition tester), but we can throw an error when we see a situation that violates the constraint.

Reply:
Ok. Another option would be to remove operator= to avoid this problem. Thoughts?

Reply:
I think that trying to set this kind of object from multiple threads is rare. Most of the time, one would set it in the constructor of a class, so that a variable can be computed in the background. `operator=` is exactly what one needs in this case.